generator.rb

```diff
@@ -0,0 +1,59 @@
+def is_vowel?(letter)
+  case letter
+  when "a", "e", "i", "o", "u"
+    letter = true
+  else
+    letter = false
+  end
+  return letter
+end
+
+def run_generator
+  puts "Please enter your first word for the portmanteau. It must be at least 2 characters long."
+  word_A = gets.chomp
+  until word_A.length >= 2
+    puts "I'm sorry, this word is invalid. Please enter a word that is at least 2 characters long."
+    word_A = gets.chomp
+  end
+  word_A_chop = 0
+  word_A_last = word_A.length - 1
+  word_A.reverse.each_char do |c|
+    if !is_vowel?(c) && word_A_chop == word_A_last #Ensures words with no vowels will include the full word in portmanteau.
+      break
+    elsif is_vowel?(c)
+      break
+    end
+    word_A_chop += 1
+  end
+  if word_A_chop == word_A_last #Ensures words with no vowels will include the full word in portmanteau.
+    puts "The first half of your word is #{word_A}."
+  else
+    word_A = word_A[0...(word_A_last - word_A_chop)]
+    puts "The first half of your word is #{word_A}."
+  end
+  puts "Please enter your second word for the portmanteau."
+  word_B = gets.chomp
+  until word_B.length >= 2
+    puts "I'm sorry, this word is invalid. Please enter a word that is at least 2 characters long."
+    word_B = gets.chomp
```

Comment:
Your code to get the second word is almost exactly the same as the code to get the first word. Could you DRY this up using another method?